### PR TITLE
Feature/reset on structure change

### PR DESCRIPTION
**Pull request by timmutton**

### CODE REVIEW

**Comment by timmutton** [comment]
@krutigarga @AlgoRyan 

I've made a change to neanderthal to actually track the structure of the configuration class and reset itself if it changes. I would like to eventually add data migration, but for now this stops it from crashing

**Comment by timmutton** [comment]
I'll throw in @lucas-outware & @huan-nguyen too, whoever is the first two to get time to do this PR

File: neanderthal/src/main/kotlin/au/com/outware/neanderthal/data/repository/VariantSharedPreferencesRepository.kt [hunk 1]
```diff
@@ -29,19 +31,32 @@ class VariantSharedPreferencesRepository(val klass: Class<out Any>,
...
-            editor.putStringSet(VARIANT_LIST, baseVariants.keys)
-            for(variant in baseVariants) {
-                editor.putString(variant.key, gson.toJson(variant.value))
-            }
-            editor.putString(CURRENT_VARIANT, defaultVariant)
-            editor.apply()
+            initialiseRepository(baseVariants, defaultVariant, structure)
+        } else if(!sharedPreferences.getStringSet(VARIANT_STRUCTURE, structure).equals(structure)) {
+            editor.clear()
+            initialiseRepository(baseVariants, defaultVariant, structure)
```
Comment (lucas-outware) [inline neanderthal/src/main/kotlin/au/com/outware/neanderthal/data/repository/VariantSharedPreferencesRepository.kt]: Could you not just simplify this to
if(!sharedPreferences.getStringSet(VARIANT_STRUCTURE, structure).equals(structure)){
editor.clear();
}
initialiseRepository(baseVariants, defaultVariant, structure);

or something close to anyway
??

Comment (timmutton) [inline neanderthal/src/main/kotlin/au/com/outware/neanderthal/data/repository/VariantSharedPreferencesRepository.kt]: not quite, although I did just clean it up a little

**Pull request merged**